internal/translation/deployment/conversion.go

```diff
@@ -0,0 +1,96 @@
+package deployment
+
+import "go.mongodb.org/atlas-sdk/v20231115008/admin"
+
+type Connection struct {
+	Name             string
+	ConnURL          string
+	SrvConnURL       string
+	PrivateURL       string
+	SrvPrivateURL    string
+	Serverless       bool
+	PrivateEndpoints []PrivateEndpoint
+}
+
+type PrivateEndpoint struct {
+	URL       string
+	ServerURL string
+	ShardURL  string
+}
+
+func clustersToConnections(clusters []admin.AdvancedClusterDescription) []Connection {
+	conns := []Connection{}
+	for _, c := range clusters {
+		conns = append(conns, Connection{
+			Name:             c.GetName(),
+			ConnURL:          c.ConnectionStrings.GetStandard(),
+			SrvConnURL:       c.ConnectionStrings.GetStandardSrv(),
+			PrivateURL:       c.ConnectionStrings.GetPrivate(),
+			SrvPrivateURL:    c.ConnectionStrings.GetPrivateSrv(),
+			Serverless:       false,
+			PrivateEndpoints: fillClusterPrivateEndpoints(c.ConnectionStrings.GetPrivateEndpoint()),
+		})
+	}
+	return conns
+}
+
+func fillClusterPrivateEndpoints(cpeList []admin.ClusterDescriptionConnectionStringsPrivateEndpoint) []PrivateEndpoint {
+	pes := []PrivateEndpoint{}
+	if len(cpeList) == 0 {
+		return pes
+	}
+	for _, cpe := range cpeList {
+		pes = append(pes, PrivateEndpoint{
+			URL:       cpe.GetConnectionString(),
+			ServerURL: cpe.GetSrvConnectionString(),
+			ShardURL:  cpe.GetSrvShardOptimizedConnectionString(),
+		})
+	}
+	return pes
+}
+
+func serverlessToConnections(serverless []admin.ServerlessInstanceDescription) []Connection {
+	conns := []Connection{}
+	for _, s := range serverless {
+		conns = append(conns, Connection{
+			Name:             s.GetName(),
+			ConnURL:          "",
+			SrvConnURL:       s.ConnectionStrings.GetStandardSrv(),
+			Serverless:       true,
+			PrivateEndpoints: fillServerlessPrivateEndpoints(s.ConnectionStrings.GetPrivateEndpoint()),
+		})
+	}
+	return conns
+}
+
+func fillServerlessPrivateEndpoints(cpeList []admin.ServerlessConnectionStringsPrivateEndpointList) []PrivateEndpoint {
+	pes := []PrivateEndpoint{}
+	if len(cpeList) == 0 {
```

Comment:
not a blocker: this feels inconsistent, some methods have this, some don't.

Reply:
will revisit

Reply:
Fixed on next update